pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,189 @@
+		if err := h.getCredentialsFromBody(writer, req, capa); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.InvalidBodyContent.Status, err.Error())
+			return
+		}
+	} else {
+		util.ReturnHTTPError(writer, req, httperror.Unauthorized.Status, "cannot access Google API without credentials to authenticate")
+	}
+
+	var serialized []byte
+	var errCode int
+	var err error
+
+	resourceType := mux.Vars(req)["resource"]
+
+	switch resourceType {
+	case "gkeMachineTypes":
+		if serialized, errCode, err = listMachineTypes(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting machine types: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeNetworks":
+		if serialized, errCode, err = listNetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting networks: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeServiceAccounts":
+		if serialized, errCode, err = listServiceAccounts(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting serviceaccounts: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeSubnetworks":
+		if serialized, errCode, err = listSubnetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting subnetworks: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeVersions":
+		if serialized, errCode, err = listVersions(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting versions: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeZones":
+		if serialized, errCode, err = listZones(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting zones: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	default:
+		util.ReturnHTTPError(writer, req, httperror.NotFound.Status, "invalid endpoint "+resourceType)
+	}
+}
+
+func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credID string) error {
+	ns, name := ref.Parse(credID)
+	if ns == "" || name == "" {
+		logrus.Debugf("[GKE] invalid cloud credential ID %s", credID)
+		return fmt.Errorf("invalid cloud credential ID %s", credID)
+	}
+
+	var accessCred client.CloudCredential //var to check access
+	if err := access.ByID(h.generateAPIContext(req), &schema.Version, client.CloudCredentialType, credID, &accessCred); err != nil {
+		if apiError, ok := err.(*httperror.APIError); ok {
+			if apiError.Code.Status == httperror.PermissionDenied.Status || apiError.Code.Status == httperror.NotFound.Status {
+				return fmt.Errorf("cloud credential not found")
+			}
+		}
+		return err
```

Comment:
I have added `handleError` function

Reply:
handleErr isn't used here. You don't want to return the actual err as it could give away information to user. Lets just follow precedent of oci and do 404: https://github.com/rancher/rancher/blob/master/pkg/api/norman/customization/oci/handler.go#L152

Reply:
now, it will return `return httperror.NotFound.Status, err` which will be passed to `handleErr()`